robot/stream/octet.robot

```diff
@@ -0,0 +1,81 @@
+*** Settings ***
+Documentation       A test suite for examples/octet_stream/key-value-store 
+
+Library             Process
+Library             String
+Library             OperatingSystem
+
+Test Setup          Start Key Value Store Server
+Test Teardown       Stop Server And Clients
+
+
+*** Variables ***
+${SERVER_HOST}      localhost
+${SERVER_PORT}      55520
+${SERVER_PATH}      /path/to/the/server
+${CLIENT_PATH}      /path/to/the/client
+
+${FIRST_CLIENT_INPUT}         { "type": "put", "key": "A", "value": "26" }\n{ "type": "get", "key": "A" }\n{ "type": "del", "key": "A" }\n{ "type": "get", "key": "A" }     
+${FIRST_CLIENT_BASELINE}      reply: \nreply: 26\nreply: 26\nreply: {"error":"no_such_key"}
+${SECOND_CLIENT_INPUT}        { "type": "put", "key": "B", "value": "27" }\n{ "type": "get", "key": "B" }\n{ "type": "del", "key": "B" }\n{ "type": "get", "key": "B" }      
+${SECOND_CLIENT_BASELINE}     reply: \nreply: 27\nreply: 27\nreply: {"error":"no_such_key"}    
+
+
+*** Test Cases ***
+Send Some message
+    Start Text Client  ${FIRST_CLIENT_INPUT}
+    Wait For Client    ${1}
+    Wait Until Contains Baseline  ${FIRST_CLIENT_BASELINE}  text-store.out
+    Start Text Client  ${SECOND_CLIENT_INPUT}
+    Wait For Client    ${2}
+    Wait Until Contains Baseline  ${SECOND_CLIENT_BASELINE}  text-store.out
+
+
+*** Keywords ***
+Start Key Value Store Server
+    Start Process
+    ...  ${SERVER_PATH}
+    ...  -p    ${SERVER_PORT}
+    ...  --caf.logger.file.path  server.log
+    ...  stdout=server.out
+    ...  stderr=server.err
+    Wait For Server Startup
+
+Start Text Client
+    [Arguments]   ${stdin}
+    Start Process
+    ...  ${CLIENT_PATH}
+    ...  -p    ${SERVER_PORT}
+    ...  --caf.logger.file.path  text-store.log
+    ...  stdin=${stdin}
```

Comment:
Is this closing `stdin` of the process after everything has been written? That would make the tests unstable, because our reader shuts down once `stdin` has been closed.

Reply:
The process does not quit after `stdin` has completed writing. We have written a teardown function which closes the running processes after the test is completed.

Reply:
Robot does close stdin of the process after writing. The example just didn't shut down as a result. I've restructured the code a bit and had to work around this behavior in the Robot tests. I've also simplified the example a bit, so please take a look at the changes I've made for future reference.